scripts/non_workflow/databundle_cli.py

```diff
@@ -0,0 +1,207 @@
+        )
+
+        # Filter out rows where the 'file path' contains '*'
+        df = df[~df.index.str.contains(r"\*", na=False)]
+        df = df[~df.index.str.endswith("/")]
+
+        df_file = pd.concat([df_file, df])
+
+    df_file["retrieve"] = [os.path.isfile(f"{i}") for i in df_file.index]
+
+    console_table(
+        df_file, table_kw={"title": "Databundle Checklist"}
+    )  # [["databundle","retrieve"]]
+
+    missing_bundles = list(df_file.loc[~df_file.retrieve, "databundle"].unique())
+
+    if len(missing_bundles) == 0:
+        console_markdown("No data is missing")
+
+    else:
+        extractable_bundle = [
+            bundlename
+            for bundlename in missing_bundles
+            if set(config_databundles[bundlename]["urls"].keys()).intersection(
+                ["zenodo", "gdrive", "direct"]
+            )
+        ]
+
+        console_markdown(
+            f"""It looks like there are missing files from: **{", ".join(missing_bundles)}**"""
+        )
+
+        if extractable_bundle:
+            df_link = pd.DataFrame()
+
+            for bundlename in extractable_bundle:
+                df = (
+                    pd.DataFrame(config_databundles[bundlename]["urls"].items())
+                    .rename(columns={0: "host", 1: "urls"})
+                    .assign(databundle=bundlename)
+                )
+
+                df.loc[df.index[1:], "databundle"] = ""
+                df = df.set_index("databundle")
+
+                df_link = pd.concat([df_link, df])
+
+            console_table(df_link, table_kw={"title": "Retrievable Databundle"})
+
+            console_markdown(
+                f"""The databundle **{", ".join(extractable_bundle)}** can be retrieved either automatically or manually downloaded."""
+            )
+
+        unextractable_bundle = [
+            bundlename
+            for bundlename in missing_bundles
+            if bundlename not in extractable_bundle
+        ]
+
+        if unextractable_bundle:
+            console_markdown(
+                f"""The databundle **{", ".join(unextractable_bundle)}** can only be retrieved automatically."""
+            )
+
+    console_markdown(
+        f"""
+    Options:
+
+    - **check**: update the checklist table to see if the file is now included
+    - **all**: get all missing databundles: **{", ".join(missing_bundles)}**
+    - **rerun**: get all databundles again: **{", ".join(bundles_to_download)}**
+    - **bundle_...**: get selected databundles, can be more than one
+    - Press **ENTER** to exit
+    """
+    )
+
+    return missing_bundles
+
+
+if __name__ == "__main__":
+
+    console_markdown(
+        """
+
+    # PyPSA-Earth Databundle Retrieval Command-Line Interface (CLI)
+
+    Find the missing file for your snakemake run and solve them your own way.
+    To prevent snakemake from trying to re-download the file again during the workflow, make sure to:
+
+    - Set `retrieve_databundle` as false
+
+    """
+    )
+
```

Comment:
To be on the safe side when importing snakemake object with `mock_snakemake`, we can use the following trick:

```suggestion
    if "snakemake" not in globals():
        from _helpers import mock_snakemake
```

That will keep snakemake object it it has been still created in any other way